dask_bigquery/core.py

```diff
@@ -0,0 +1,260 @@
+from __future__ import annotations
+
+import logging
+from collections.abc import Iterable
+from contextlib import contextmanager
+from functools import partial
+
+import pandas as pd
+import pyarrow
+from dask.base import tokenize
+from dask.dataframe.core import new_dd_object
+from dask.highlevelgraph import HighLevelGraph
+from dask.layers import DataFrameIOLayer
+from google.cloud import bigquery, bigquery_storage
+
+
+@contextmanager
+def bigquery_client(project_id=None, with_storage_api=False):
+    """This context manager is a temporary solution until there is an
+    upstream solution to handle this.
+    See  googleapis/google-cloud-python#9457
+    and googleapis/gapic-generator-python#575 for reference.
+    """
+
+    bq_storage_client = None
+    bq_client = bigquery.Client(project_id)
```

Comment:
From what I understand, now we do have a context manager for the bigquery client but not for the bigquery_storage_client. Would you suggest using at least the one we have?  Something like
```python
bq_storage_client = None
with bigquery.Client(project_id) as bq_client:
    try:
        if with_storage_api:
            bq_storage_client = bigquery_storage.BigQueryReadClient(
                credentials=bq_client._credentials
            )
            yield bq_client, bq_storage_client
        else:
            yield bq_client
```

Reply:
Yep, that's the right direction. Though since `bq_client.close()` is all that was being called in the `finally` block before, we can remove the `try`/`finally` blocks since `bq_client.close()` will be called when we exit the scope of the `bq_client` context manager

Reply:
Since we have to include the closing of the storage client, maybe we should keep the try/finally but on the finally have  the `bigquery_storage.transport.channel.close()`